server/api/search.ts

```diff
@@ -3,45 +3,68 @@ import { serverQueryContent } from '#content/server'
 
 export default defineEventHandler(async (event) => {
     // Fetch all documents
-    let docs = await serverQueryContent(event).find()
-
-    docs = await Promise.all(
-        docs
-            .filter(
-                (doc) => {
-                    return doc?._extension === 'md' &&
-                        doc?._draft === false &&
-                        doc?.hidden === true &&
-                        !doc?._empty
-                }
-            )
-            .map(
-                async ({ _id: id, _path: path, _dir: dir, title = '', description = '', body = undefined}) => {
-
-                    return {
-                        id,
-                        path,
-                        dir,
-                        title,
-                        description,
-                        keywords: body?.toc?.links.map(link => link?.text),
-                        body: extractTextFromAst(body) || ''
-                    }
-                }
-            )
+    const docs = (await serverQueryContent(event).where({$and: [
+        {
+            _extension: { $eq: "md" },
+        },
+        {
+            _partial: { $eq: false },
+        },
+        {
+            _draft: { $ne: true },
+        },
+        {
+            hidden: { $ne: true },
+        }
+    ]}).find()).filter(
+        (doc) => {
+            // Remove empty docs
+            return doc.body?.children.length !== 0;
+        }
     )
 
-    return docs
+    // TODO: Allow body limit customisation
```

Comment:
indeed, it could an advanced option in app.config.ts

Reply:
Agreed, will want separate PR as this one is fixes only.